Fig.Core/CompositeSettingsDictionary.cs

```diff
@@ -54,6 +55,19 @@ private string GetCurrentEnvironment(Match m, string configuration)
             TryGetValue(key, env, out var result);
             return result;
         }
-
+        private bool ConcatIndices(SettingsDictionary sd, string key, out string value)
+        {
+            int index = 0;
+            string concatValue = null;
+            value = null;
+            while (sd.TryGetValue(key + "." + index, out value))
+            {
+                index++;
+                concatValue += value + ",";
+            }
+            if (concatValue != null) concatValue = concatValue.TrimEnd(',');
```

Comment:
If you use `String.Join` instead then you don't have to pull off the extra comma after the loop. It will also open up for some additional simplification.

Reply:
I've simplified the code to use string.Join, the automated test failed but on my local branch its green, 'preload Settings.cs:line 449' odd because mine only contains 373 lines.

Reply:
I recently merged a PR so you need to pull and integrate

Reply:
Should be resolved now.